### PR TITLE
Rerun integration test with `LogLevel::Trace` if `RUST_LOG` is not set manually

### PR DESCRIPTION
**Pull request by TimDiekmann**
## 🌟 What is the purpose of this PR?

If our tests are failing, we want to know why. As always running them with `RUST_LOG=trace` affects performance, they should run with a lower severity unless it fails.

## 🔍 What does this change?

- Adds a command-line argument to specify the `LogLevel`
- Runs tests with `LogLevel::Warning` and rerun it with `LogLevel::Trace` if it failed
- If `RUST_LOG` is set manually, only one attempt is running

### CODE REVIEW

**Review by Alfred-Mountfield — APPROVED** [review]

File: packages/engine/tests/experiment/mod.rs [hunk 1]
```diff
@@ -160,35 +163,55 @@ pub async fn run_test_suite(
...
-            let test_result = match test_result {
-                Ok(outputs) => outputs,
-                Err(err) => {
-                    match fs::read_to_string(&log_file_path) {
-                        Ok(log) => eprintln!("{log}"),
-                        Err(err) => eprintln!("Could not read {log_file_path:?}: {err}"),
+            let mut test_output = None;
+            for rust_log_env in &rust_log_envs {
+                let environment = std::env::vars();
+                std::env::set_var("RUST_LOG", &rust_log_env);
```
Comment (Alfred-Mountfield) [inline packages/engine/tests/experiment/mod.rs:169]: I think we have to somehow manually pass in the log level. Using env-vars won't work as they're within the same process

Comment (TimDiekmann) [reply packages/engine/tests/experiment/mod.rs]: I added a command-line argument and use this instead.

File: packages/engine/lib/orchestrator/src/experiment.rs [hunk 2]
```diff
@@ -87,8 +91,8 @@ pub struct ExperimentConfig {
     ///
     /// Defaults to the number of logical CPUs available in order to maximize performance.
     #[cfg_attr(
-        feature = "clap",
-        clap(global = true, short = 'w', long, default_value_t = num_cpus::get(), validator = at_least_one, env = "HASH_WORKERS")
```
Comment (TimDiekmann) [inline packages/engine/lib/orchestrator/src/experiment.rs:91]: `rustfmt` is weird sometimes...

File: packages/engine/lib/orchestrator/src/experiment.rs [hunk 3]
```diff
@@ -176,7 +181,7 @@ impl Experiment {
     /// returns once the experiment has finished.
     ///
     /// [`Process`]: crate::process::Process
-    #[instrument(skip_all, fields(experiment_name = %experiment_run.base.name, experiment_id = %experiment_run.base.id))]
+    #[instrument(skip_all, fields(experiment_name = % experiment_run.base.name, experiment_id = % experiment_run.base.id))]
```
Comment (Alfred-Mountfield) [inline packages/engine/lib/orchestrator/src/experiment.rs:184]: Was this rustfmt? I don't think there should be spaces here. There aren't in the tracing docs

**Review by Alfred-Mountfield — COMMENTED** [review]
It's a shame it adds so much boilerplate. Perhaps we can cut down on it at some point and make it cleaner.

File: packages/engine/lib/orchestrator/src/experiment.rs (hunk 3 above)
Comment (TimDiekmann) [reply packages/engine/lib/orchestrator/src/experiment.rs:184]:
```suggestion
    #[instrument(skip_all, fields(experiment_name = %experiment_run.base.name, experiment_id = % experiment_run.base.id))]
```

File: packages/engine/lib/orchestrator/src/experiment.rs [hunk 4]
```diff
@@ -176,7 +181,7 @@ impl Experiment {
     /// returns once the experiment has finished.
     ///
     /// [`Process`]: crate::process::Process
-    #[instrument(skip_all, fields(experiment_name = %experiment_run.base.name, experiment_id = %experiment_run.base.id))]
+    #[instrument(skip_all, fields(experiment_name = %experiment_run.base.name, experiment_id = % experiment_run.base.id))]
```
Comment (Alfred-Mountfield) [inline packages/engine/lib/orchestrator/src/experiment.rs:184]:
```suggestion
    #[instrument(skip_all, fields(experiment_name = %experiment_run.base.name, experiment_id = %experiment_run.base.id))]
```

**Review by Alfred-Mountfield — CHANGES_REQUESTED** [review]

**Review by Alfred-Mountfield — DISMISSED** [review]

**Comment by Alfred-Mountfield** [comment]
> Talking about the tests in general
I mean the boilerplate around redefining Log Levels and mirroring the tracing impl, etc.

**Comment by TimDiekmann** [comment]
> > Talking about the tests in general
> 
> I mean the boilerplate around redefining Log Levels and mirroring the tracing impl, adding more CLI args, etc.

Yeah, this is pretty annoying. 😒

**Review by Alfred-Mountfield — APPROVED** [review]

**Pull request merged**